ssas/service/logging.go

```diff
@@ -13,18 +13,19 @@ import (
 	"github.com/CMSgov/bcda-ssas-app/ssas"
 )
 
-// https://github.com/go-chi/chi/blob/master/_examples/logging/main.go
+//https://github.com/go-chi/chi/blob/master/_examples/logging/main.go
 
 func NewAPILogger() func(next http.Handler) http.Handler {
 	return middleware.RequestLogger(&APILogger{ssas.Logger})
 }
 
 type APILogger struct {
-	Logger *logrus.Logger
+	Logger logrus.FieldLogger
 }
 
 func (l *APILogger) NewLogEntry(r *http.Request) middleware.LogEntry {
-	entry := &APILoggerEntry{Logger: logrus.NewEntry(l.Logger)}
+	// change the parameter to take an entry for fieldLogger, change line 23 to entry, pass in an entry
+	entry := &APILoggerEntry{Logger: l.Logger}
```

Comment:
similar question to the above — is there a difference between using the existing logger vs. creating a new entry?

Reply:
this was an odd hurdle; creating a New Entry will remove the env, app, and version fields. when I ran it locally, it didn't merge all of the fields, but I can do some additional testing to make sure there's no off cases where that might happen. Another alternative would be to create a logger instance for the middleware and one for the operations.

Reply:
when we create log fields and use `withField` or `withFields`, it creates a new Entry object, so this entry for the middleware and the entry for the operations should stay separate. Using either of those functions doesn't attribute the fields to the logger, but the returned Entry can be used to persist fields during log calls -- seems a little counterintuitive. The use of FieldLogger generalizes Entry and Logger which is how we can pass around the instance.

Reply:
thanks for the detailed explanation! The fact that FieldLogger is a generalization of those two interfaces helps to understand the changes 😄